var/spack/repos/builtin/packages/mathematica/package.py

```diff
@@ -31,6 +32,12 @@ class Mathematica(Package):
     license_url      = 'https://reference.wolfram.com/language/tutorial/RegistrationAndPasswords.html#857035062'
 
     def install(self, spec, prefix):
+        # Backup .spack because Mathematica moves it but never restores it
+        rand_suffix = random.randint(1, 65536)
+        cp = which('cp')
+        cp('-a', '{0}/.spack'.format(prefix),
+           '/tmp/.spack-{0}'.format(rand_suffix))
```

Comment:
I would use Spack's `copy()` method for this. Same below. Also, if you copy it to the stage directory, you won't need a random suffix. Is there even anything in `<prefix>/.spack` before the installation starts? Maybe you can just create a directory at the end. Does the build log still get copied?

Reply:
@adamjstewart thanks for the pointers.

The `.spack` directory already contains the `spack-build-*.txt` files. All files are there when installation ends. I think there's some Linux magic under the hood.

I'll use `copy()` to move the directory to the stage directory.